gateway-server/src/main/java/org/apache/knox/gateway/services/topology/impl/DefaultTopologyService.java

```diff
@@ -556,6 +553,11 @@ public void startMonitor() throws Exception {
         log.remoteConfigurationMonitorStartFailure(remoteMonitor.getClass().getTypeName(), e.getLocalizedMessage());
       }
     }
+
+    // Trigger descriptor discovery (KNOX-2301)
+    for (File descriptor : getDescriptors()) {
+      descriptor.setLastModified(System.currentTimeMillis()); // 'Touch' the descriptor
+    }
```

Comment:
I'd move this out to a new method instead of keeping it in `startMonitor` with a good method name. Otherwise, in the future, it'd confuse others why it's here.

Reply:
This is the reason for the comment, but pulling it out into a separate explicitly-named method is more clear and resilient to future confusion from the comment not being updated.